Devall/Tabatadze/Controller/Company/EditPost.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+namespace Devall\Tabatadze\Controller\Company;
+
+use Devall\Tabatadze\Model\Company;
+use Exception;
+use Magento\Customer\Api\Data\CustomerInterface;
+use Magento\Customer\Model\AddressRegistry;
+use Magento\Framework\Data\Form\FormKey\Validator;
+use Magento\Customer\Model\Session;
+use Magento\Framework\App\Action\Context;
+use Magento\Framework\Exception\NoSuchEntityException;
+use Magento\Framework\View\Result\PageFactory;
+use Magento\Framework\Controller\ResultFactory;
+use Magento\Customer\Api\CustomerRepositoryInterface;
+use Magento\Framework\App\Action\Action;
+
+class EditPost extends Action
+{
+    /**
+     * @var Session
+     */
+    protected $session;
+
+    /**
+     * @var PageFactory
+     */
+    protected $resultPageFactory;
+
+    /**
+     * @var Validator
+     */
+    protected $formKeyValidator;
+
+    /**
+     * @var CustomerRepositoryInterface
+     */
+    private $customerRepositoryInterface;
+
+    /**
+     * @var AddressRegistry
+     */
+    private $addressRegistry;
+
+    const COMPANY_KEY = 'company';
+    const DOMAIN_NAME = 'http://localhost';
```

Comment:
You need to find a way to get the hosts url dynamically. So that if I install this extension on my computer or a client installs it on a live site, we get the correct url without having to change the constant every time. Or if we have multiple websites with different urls, it should also work.